app/models/ssh_execution_provider.rb

```diff
@@ -51,5 +51,15 @@ def ssh_user(host)
     def ssh_port(host)
       Integer(host_setting(host, :remote_execution_ssh_port))
     end
+
+    def ssh_proxy_params(proxy)
+      return proxy if proxy.class == Symbol
+
+      {
+        :url => proxy.url,
+        :ssl_certificate => File.open(Setting[:ssl_certificate]).read,
+        :ssl_priv_key => File.open(Setting[:ssl_priv_key]).read
```

Comment:
The opened `File`s need to be closed. After a while the proxy would crash because of reaching the opened file limit. Something like `File.open(Setting[:ssl_priv_key], &:read)` might do the trick

Reply:
Or even simpler `File.read`

Reply:
Oh, does the Ruby GC not close files?

Reply:
We should catch exceptions here, and probably only add this information when the proxy uses https.

Reply:
Actually, we should perhaps not return this at all in params, but rather configure the session script to read the right files form the filesystem: returning this via http might have potencial recurity risks that we should avoid